crates/matrix-sdk/src/client/mod.rs

```diff
@@ -444,6 +444,13 @@ impl Client {
         self.base_client().session_meta()
     }
 
+    /// Returns a receiver that gets events for each room info update. To watch
+    /// for new events, use `receiver.resubscribe()`. Each event contains the
+    /// room and a boolean whether this event should trigger a room list update.
+    pub fn roominfo_update_receiver(&self) -> &broadcast::Receiver<RoomInfoUpdate> {
+        self.base_client().roominfo_update_receiver()
+    }
+
```

Comment:
I think we could get rid of this method: `base_client()` is public on the `Client`, so it's possible to get the `BaseClient::roominfo_update_receiver` that way, and it's kind of nice that it's semi-hidden by not having it in the public API, since this is a bit of an internal implementation detail. Can you remove it please?

Reply:
base_client is only pub(crate) unfortunately